test/functional/sqlsrv/sqlsrv_ae_output_param_sqltype_binary_size.phpt

```diff
@@ -0,0 +1,279 @@
+--TEST--
+Test for retrieving encrypted data of binary types with different sizes as output parameters
+--DESCRIPTION--
+Test implicit conversions between different binary types of different sizes
+With Always Encrypted, implicit conversion works if:
+1. From a binary(m) column to a SQLSRV_SQLTYPE_BINARY(n) output parameter where m == n
+2. From a binary(m) column to a SQLSRV_SQLTYPE_VARBINARY(n) output parameter where m == n
+3. From a varbinary(m) column to a SQLSRV_SQLTYPE_BINARY(n) output parameter where m == n
+4. From a varbinary(m) column to a SQLSRV_SQLTYPE_VARBINARY(n) output parameter where m == n
+Without AlwaysEncrypted, implicit conversion works if:
+1. From a binary(m) column to a SQLSRV_SQLTYPE_BINARY(n) output parameter where m, n == any value
+2. From a binary(m) column to a SQLSRV_SQLTYPE_VARBINARY(n) output parameter where m <= n (exclude SQLSRV_SQLTYPE_VARBINARY('max'))
+3. From a varbinary(m) column to a SQLSRV_SQLTYPE_BINARY(n) output parameter where m, n == any value
+4. From a varbinary(m) column to a SQLSRV_SQLTYPE_VARBINARY(n) output parameter where m, n == any value
+--SKIPIF--
+<?php require('skipif_versions_old.inc'); ?>
+--FILE--
+<?php
+require_once('MsCommon.inc');
+
+$dataTypes = array("binary", "varbinary", "varbinary(max)");
+$lengths = array(1, 8, 64, 512, 4000);
+$sqlTypes = array("SQLSRV_SQLTYPE_BINARY", "SQLSRV_SQLTYPE_VARBINARY", "SQLSRV_SQLTYPE_VARBINARY('max')");
+$sqltypeLengths = $lengths;
+$directions = array("SQLSRV_PARAM_OUT", "SQLSRV_PARAM_INOUT");
+$inputValues = array("d", "f");
+
+$conn = AE\connect();
+foreach($dataTypes as $dataType) {
+    $maxcol = strpos($dataType, "(max)");
+    foreach($lengths as $m) {
+        if ($maxcol !== false) {
+            $typeFull = $dataType;
+        } else {
+            $typeFull = "$dataType($m)";
+        }
+        echo "\nTesting $typeFull:\n";
+            
+        // create and populate table containing binary(m) or varbinary(m) columns
+        $tbname = "test_" . str_replace(array('(', ')'), '', $dataType) . $m;
+        $colMetaArr = array(new AE\ColumnMeta($typeFull, "c_det"), new AE\ColumnMeta($typeFull, "c_rand", null, false));
+        AE\createTable($conn, $tbname, $colMetaArr);
+        $inputs = array(new AE\BindParamOption($inputValues[0], null, "SQLSRV_PHPTYPE_STRING(SQLSRV_ENC_BINARY)"),
+                                new AE\BindParamOption($inputValues[1], null, "SQLSRV_PHPTYPE_STRING(SQLSRV_ENC_BINARY)"));
+        $r;
+        $stmt = AE\insertRow($conn, $tbname, array("c_det" => $inputs[0], "c_rand" => $inputs[1]), $r, AE\INSERT_PREPARE_PARAMS);
```

Comment:
instead of "c_det" and "c_rand" use $colMetaArr[0]->colName and $colMetaArr[1]->colName?